### PR TITLE
Add some minor functionality to FunctionTreeVector

### PR DESCRIPTION
**Pull request by ilfreddy**
I upgraded `FunctionTreeVector`with some minor functionality:
1) `push_back` a whole `FunctionTreeVector`
2) count the total number of nodes.

The functionality is IMO generic enough, and in particular it allows me to simplify `XCFunctional` and to remove some duplicate stuff there.

### CODE REVIEW

**Comment by robertodr** [comment]
Would it make sense to implement this using STL containers? This seems to be a `std::vector<std::tuple<double, FunctionTree<D> *>>` You'd probably get a lot of the functions implemented for free.

**Comment by stigrj** [comment]
Good idea @robertodr

File: src/trees/FunctionTreeVector.h [hunk 1]
```diff
@@ -17,9 +17,11 @@ class FunctionTreeVector {
 
     int size() const { return this->funcs.size(); }
     void clear(bool dealloc = false);
+    int sumNodes();
```
Comment (stigrj) [inline src/trees/FunctionTreeVector.h]: I think you can add a trailing `const` on this one

**Comment by stigrj** [comment]
I'm squashing this one

**Pull request merged**